app/controllers/service_types_controller.rb

```diff
@@ -3,8 +3,40 @@ class ServiceTypesController < ApplicationController
 
   # Authentication by default inherited from ApplicationController.
 
+  before_action :authorize_for_districtwide_access_admin, except: [:index]
+
+  def authorize_for_districtwide_access_admin
+    unless current_educator.admin? && current_educator.districtwide_access?
+      error = 'You don\'t have the correct authorization.'
+
+      return render json: { error: error }, status: 403
+    end
+  end
+
   def index
     render json: ServiceType.pluck(:name).sort
   end
 
+  def is_service_working_json
+    service_type_id = params[:service_type_id]
+
+    service_type = ServiceType.find(service_type_id)
+    student_ids = service_type.services.where('date_started > ?', Time.current - 1.year).map(&:student_id)
```

Comment:
I think this is a good place to use `authorized` as an extra layer.

```
students = authorized do
  service_type.services.where('date_started > ?', Time.current - 1.year).map(&:students)
end
```

That will bounce each student off of `authorizer.is_authorized_for_student?` (the block is just stylistic, it functions the same as if you passed an argument).

The benefit of moving the authorization check close to the query is that if the code is refactored and the controller guard changes, the query itself has its own auth check already.